src/Compilers/CSharp/Portable/Symbols/LocalSymbol.cs

```diff
@@ -25,6 +25,11 @@ protected LocalSymbol()
             get;
         }
 
+        /// <summary>
+        /// Syntax node that is used as the scope designator. Otherwise, null.
+        /// </summary>
+        internal abstract SyntaxNode ScopeDesignatorOpt { get; }
```

Comment:
These are user-defined locals regardless of where they are declared, they are not synthesized. I do not believe it is appropriate to change SynthesizedLocalKind for them.